### PR TITLE
🚀 Add missing event values in crafting events + fix `PrepareItemCraftEvent`

### PR DESCRIPTION
**Pull request by AyhamAl-Ali**
### Description

This PR aims to add missing event-values to prepare crafting and craft events `event-inventory event-itemtype` 
This PR also fixes a bug in `PrepareItemCraftEvent` not working as expected (was running when you click the result item rather than working before the result item appear)

---
**Target Minecraft Versions:** Any
**Requirements:** None
**Related Issues:** None

### CODE REVIEW

File: src/main/java/ch/njol/skript/classes/data/BukkitEventValues.java [hunk 1]
```diff
@@ -1001,22 +997,26 @@ public Player get(BlockFertilizeEvent event) {
...
-			@Nullable
 			public Slot get(final PrepareItemCraftEvent e) {
 				return new InventorySlot(e.getInventory(), 9);
 			}
 		}, 0);
+		EventValues.registerEventValue(PrepareItemCraftEvent.class, ItemType.class, new Getter<ItemType, PrepareItemCraftEvent>() {
+			@Override
+			public ItemType get(final PrepareItemCraftEvent e) {
+				ItemStack item = e.getInventory().getResult();
+				return new ItemType(item != null ? item : new ItemStack(Material.AIR));
```
Comment (Pikachu920) [inline src/main/java/ch/njol/skript/classes/data/BukkitEventValues.java:1011]: `ItemType` has a constructor which takes a `Material`. You could avoid the `ItemStack` instantiation by splitting this into an if/else and using that constructor

**Review by Pikachu920 — CHANGES_REQUESTED** [review]

File: src/main/java/ch/njol/skript/classes/data/BukkitEventValues.java [hunk 2]
```diff
@@ -20,6 +20,7 @@
 
 import java.util.List;
 
+import ch.njol.skript.util.*;
```
Comment (TheLimeGlass) [inline src/main/java/ch/njol/skript/classes/data/BukkitEventValues.java:23]: Please disable wild card imports in your IDE. We can't be importing classes we don't need.

File: src/main/java/ch/njol/skript/classes/data/BukkitEventValues.java [hunk 3]
```diff
@@ -1001,22 +997,26 @@ public Player get(BlockFertilizeEvent event) {
...
 		EventValues.registerEventValue(PrepareItemCraftEvent.class, Slot.class, new Getter<Slot, PrepareItemCraftEvent>() {
 			@Override
-			@Nullable
 			public Slot get(final PrepareItemCraftEvent e) {
 				return new InventorySlot(e.getInventory(), 9);
 			}
 		}, 0);
+		EventValues.registerEventValue(PrepareItemCraftEvent.class, ItemType.class, new Getter<ItemType, PrepareItemCraftEvent>() {
+			@Override
+			public ItemType get(final PrepareItemCraftEvent e) {
```
Comment (TheLimeGlass) [inline src/main/java/ch/njol/skript/classes/data/BukkitEventValues.java:1009]: You can just return an ItemStack. We need to be able to remember lore, names, enchantments, etc. Skript will convert an ItemStack to an ItemType if the user attempts to compare an ItemStack with an ItemType.

File: src/main/java/ch/njol/skript/classes/data/BukkitEventValues.java [hunk 4]
```diff
@@ -1053,6 +1053,14 @@ public String get(PrepareItemCraftEvent e) {
 				}
 			}, 0);
 		}
+		// CraftItemEvent
+		EventValues.registerEventValue(CraftItemEvent.class, ItemType.class, new Getter<ItemType, CraftItemEvent>() {
+			@Override
+			@Nullable
+			public ItemType get(CraftItemEvent e) {
```
Comment (TheLimeGlass) [inline src/main/java/ch/njol/skript/classes/data/BukkitEventValues.java:1060]: You can just return an ItemStack. We need to be able to remember lore, names, enchantments, etc. Skript will convert an ItemStack to an ItemType if the user attempts to compare an ItemStack with an ItemType.

File: src/main/java/ch/njol/skript/classes/data/BukkitEventValues.java [hunk 5]
```diff
@@ -1001,22 +1000,26 @@ public Player get(BlockFertilizeEvent event) {
...
+		EventValues.registerEventValue(PrepareItemCraftEvent.class, ItemStack.class, new Getter<ItemStack, PrepareItemCraftEvent>() {
+			@Override
+			public ItemStack get(final PrepareItemCraftEvent e) {
+				ItemStack item = e.getInventory().getResult();
+				return item != null ? item : AIR_IS;
+			}
+		}, 0);
+		EventValues.registerEventValue(PrepareItemCraftEvent.class, Inventory.class, new Getter<Inventory, PrepareItemCraftEvent>() {
+			@Override
+			public Inventory get(final PrepareItemCraftEvent e) {
```
Comment (TheLimeGlass) [inline src/main/java/ch/njol/skript/classes/data/BukkitEventValues.java:1019]: We have been phasing out the over usage of final in methods. Final is great practice but it's also not required or useful for security in Spigot for majority of Skript. You also didn't add it for the CraftItemEvent value, so we shouldn't be adding final or worring about it.

**Review by TheLimeGlass — CHANGES_REQUESTED** [review]

File: src/main/java/ch/njol/skript/classes/data/BukkitEventValues.java (hunk 5 above)
Comment (AyhamAl-Ali) [reply src/main/java/ch/njol/skript/classes/data/BukkitEventValues.java:1019]: Sure :+1:

**Review by TheLimeGlass — APPROVED** [review]

**Review by TheLimeGlass — APPROVED** [review]

**Review by APickledWalrus — CHANGES_REQUESTED** [review]
~~This is counter to Lime's review, but you probably should return ItemType. All other event values do, and there's no reason to potentially convert it multiple times later when it can be done once now :) (plus, there have been some errors due to inconsistencies in choosing one to return, so we like to stick with itemtype)~~

EDIT: this may not actually matter too much tbh, the points I mentioned above have likely become less of an issue (due to numerous PRs to improve behavior) - you can disregard this :)

**Review by APickledWalrus — APPROVED** [review]

**Pull request merged**